src/webapp/components/survey/SurveyForm.tsx

```diff
@@ -108,19 +118,29 @@ export const SurveyForm: React.FC<SurveyFormProps> = props => {
     };
 
     const onOrgUnitChange = (orgUnitPaths: string[]) => {
```

Comment:
Maybe you can add these type of functions in useSurveyForm and call useSaveSurvey inside the hook? I think that they are also from presentation logic